frontend/packages/console-app/src/components/detect-namespace/getValueForNamespace.ts

```diff
@@ -0,0 +1,37 @@
+import { ALL_NAMESPACES_KEY } from '@console/shared/src/constants';
+import { PREFERRED_NAMESPACE_USER_SETTING_VALUE_LATEST } from '../user-preferences/namespace';
+import { checkNamespaceExists } from './checkNamespaceExists';
+
+export const getValueForNamespace = async (
+  useProjects?: boolean,
+  urlNamespace?: string,
+  activeNamespace?: string,
+  preferredNamespace?: string,
+  lastNamespace?: string,
+): Promise<string> => {
+  if (urlNamespace) {
+    if (await checkNamespaceExists(urlNamespace, useProjects)) {
+      return urlNamespace;
+    }
```

Comment:
I was thinking that it shouldn't throw since any error thrown is caught in `checkNamespaceExists`.
Please let me know if I am missing a scenario here.

Reply:
looks like `checkNamespaceExists` will never throw. So this is fine.